documentation/guide-access-control.asciidoc

```diff
@@ -203,7 +146,58 @@ public class UcSafeOfferImpl extends ApplicationUc implements UcSafeOffer {
+* `javax.annotation.security.DenyAll` specifies that no security roles are allowed to invoke the specified method(s).
+* `javax.annotation.security.RolesAllowed` specifies that only a list of security roles are allowed to invoke the specified method(s).
+* `javax.annotation.security.DeclareRoles` defines roles for security checking.
+* `javax.annotation.security.RunAs` specifies the RunAs role for the given components.
+
+`@PermitAll` `@Denyall` `@RolesAllowed` annotations can be applied to both class and method. A method-level annotation will override the behaviour of class-level annotation. Using multiple annotations of those 3 is not valid.
+```java
+// invalid
+@PermitAll
+@DenyAll
+public String foo()
+
+// invalid and compilation fails
+@RolesAllowed("admin")
+@RolesAllowed("user")
+public String bar()
+
+// OK
+@RolesAllowed("admin", "user")
+public String bar()
+```
+
+In many complicated cases where `@PermitAll` `@DenyAll` `@RolesAllowed` are insufficient e.g. a method should be accessed by a user in role A and not in role B at the same time, you have to verify the user role directly in the method. You can use `SecurityContext` class to get further needed information.
+
+==== Spring
+Spring Security also supports authorization on method level. To use it, you need to add the `spring-security-config` dependency. If you use Spring Boot, the dependency `spring-boot-starter-security` already includes `spring-security-config`. Then you can configure as follows:
+
+    * `prePostEnabled` property enables Spring Security pre/post annotations. `@PreAuthorize` and `@PostAuthorize` annotations provide expression-based access control. See more https://docs.spring.io/spring-security/site/docs/4.2.x/reference/html/el-access.html[here]
+    * `securedEnabled` property determines if the `@Secured` annotation should be enabled. `@Secured` can be used similarly as `@RollesAllowed`.
+    * `jsr250Enabled` property allows us to use the JSR-250 annotations such as `@RolesAllowed`.
+
+```java
+@Configuration
+@EnableGlobalMethodSecurity(
+  prePostEnabled = true,
+  securedEnabled = true,
+  jsr250Enabled = true)
+public class MethodSecurityConfig
+  extends GlobalMethodSecurityConfiguration {
+}
+```
+
+A further read about the whole concept of Spring Security Authorization can be found https://docs.spring.io/spring-security/site/docs/current/reference/html5/#servlet-authorization[here].
+
+==== Quarkus
+Quarkus comes with built-in security to allow for RBAC based on the common security annotations `@RolesAllowed`, `@DenyAll`, `@PermitAll` on REST endpoints and CDI beans. Quarkus also provides the `io.quarkus.security.Authenticated` annotation that will permit any authenticated user to access the resource (equivalent to @RolesAllowed("**")).
+
+=== Data-based Permissions
 See link:guide-data-permission.asciidoc[data permissions]
 
 === Access Control Schema (deprecated)
```

Comment:
we could actually extract this to a separate old guide and just link it from here (not even include it in devon4j-doc nor link it from _Sidebar). I doubt that there are many projects out there still using this legacy approach.

Reply:
It is indeed done so. The section "Access Control Schema" does not contain any content except a link to an old guide. I will remove link to guide-access-control-schema from sidebar and devon4j-doc